docs/tutorials/underfitting_overfitting.md

```diff
@@ -0,0 +1,160 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# How to deal with underfitting and overfitting?
+
++++
+
+In this tutorial we'll cover a common challenge in ML, and how to fix it.
+We will understand what is underfitting, overfitting and the difference between the two. 
+Once we cover the basics, we will dive deep into:
+1. Detect we have an under/overfitting (and understand which one do we have).
+2. Understand how to solve it, and if our method solved it.
+
+We will see how the different plots on our disposal (via sklearn-evaluation) can shorten our time to a working model.
+
+**TLDR: A learning curve shows the performance of a model on training and validation data as the amount of data used to train the model increases.** 
+
+**This will indicate if the model is suffering from overfitting or underfitting.**
+
++++
+
+## Overfitting
+
++++
+
+Overfitting occurs when a machine learning model is trained too well on the training data.
+As a result, it performs poorly on new, unseen data. There are a few signs that a model may be overfitting:
+1. The model's performance on training data is significantly better than its performance on the validation or test data.
+2. The model's performance flattens or starts to decrease after a certain point in the training process.
+3. The model's complexity is significantly greater than the complexity of the problem.
+
+**TODO: Add correlating learning curve**
+
++++
+
+## Underfitting
+
++++
+
+Underfitting happens when our model cannot capture the underlying patterns in the training data.
+This resulting in poor performance on both the training and validation/test sets. 
+
+There are a few signs that a model may be underfitting:
+1. The model's performance on the training data is significantly worse than expected or worse than a simple baseline model.
+2. The model's performance on the validation/test data is not much better than random guessing.
+3. The model's complexity is significantly lower than the complexity of the problem.
+
+**TODO: Add correlating learning curve**
+
++++
+
+## Detection
+
+In this section we'll detect which one we're having and what does the learning curve tells us?
+
++++
+
+## How to solve it?
+
+We'll talk about a few ways we can improve our performance back. As some solutions can help with both and some are relevant only to one set of issues, we'll seperate the methods into 2 sections.
+
+
+### Overfitting
+Regularization, dropout, early stopping, cross-validation, data augmentation, ensemble methods.
+
+### Underfitting:
+Increasing the model's complexity, Adding/Collecting more data, Using more informative features, ensemble methods, hyperparameter tuning, transfer learning.
+
+
++++
+
+# Conclusion
+
+```{code-cell} ipython3
```

Comment:
looks like from here til the bottom it was copied from another file?

Reply:
Yup, I thought I removed it.